ddtrace/opentracer/tracer.py

```diff
@@ -94,7 +109,7 @@ def __init__(self, service_name=None, config=None, scope_manager=None, dd_tracer
             settings=self._config.get(keys.SETTINGS),
             priority_sampling=self._config.get(keys.PRIORITY_SAMPLING),
             uds_path=self._config.get(keys.UDS_PATH),
-            context_provider=dd_context_provider,
+            context_provider=dd_context_provider,  # type: ignore[arg-type]
```

Comment:
why this ignore?

Reply:
The ddtrace.tracer config expects the `context_provider` argument to be of type `Optional[DefaultContextProvider]`, but technically the opentracer provider can be of type `DefaultContextProvider` as well as `AsyncioContextProvider` which inherits from `BaseContextProvider` instead of `DefaultContextProvider`... it may be too messy to resolve this so I slapped the ignore here ☹️

Reply:
oof, hmm... seems better to just ignore. :/